src/Sentry/Protocol/SentryAttribute.cs

```diff
@@ -0,0 +1,107 @@
+internal readonly struct SentryAttribute
+{
+    public SentryAttribute(object value, string type)
+    {
+        Value = value;
+        Type = type;
+    }
+
+    public object Value { get; }
+    public string Type { get; }
+}
+
+internal static class SentryAttributeSerializer
+{
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, SentryAttribute attribute)
+    {
+        Debug.Assert(attribute.Value is not null && attribute.Type is not null, $"The ValueType {nameof(attribute)} may have been assigned 'default', for which static flow analysis does not report nullable warnings.");
+        writer.WritePropertyName(propertyName);
+        WriteAttributeValue(writer, attribute.Value, attribute.Type);
+    }
+
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, object value, string type)
+    {
+        writer.WritePropertyName(propertyName);
+        WriteAttributeValue(writer, value, type);
+    }
+
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, object value)
+    {
+        writer.WritePropertyName(propertyName);
+        WriteAttributeValue(writer, value);
+    }
+
+    private static void WriteAttributeValue(Utf8JsonWriter writer, object value, string type)
+    {
+        writer.WriteStartObject();
+
+        if (type == "string")
+        {
+            writer.WriteString("value", (string)value);
+            writer.WriteString("type", type);
+        }
+        else if (type == "boolean")
+        {
+            writer.WriteBoolean("value", (bool)value);
+            writer.WriteString("type", type);
+        }
+        else if (type == "integer")
+        {
+            writer.WriteNumber("value", (long)value);
+            writer.WriteString("type", type);
+        }
+        else if (type == "double")
+        {
+            writer.WriteNumber("value", (double)value);
+            writer.WriteString("type", type);
+        }
+        else
+        {
+            writer.WriteString("value", value.ToString());
+            writer.WriteString("type", "string");
+        }
+
+        writer.WriteEndObject();
+    }
+
+    private static void WriteAttributeValue(Utf8JsonWriter writer, object value)
+    {
+        writer.WriteStartObject();
+
+        if (value is string str)
+        {
+            writer.WriteString("value", str);
+            writer.WriteString("type", "string");
+        }
+        else if (value is bool boolean)
+        {
+            writer.WriteBoolean("value", boolean);
+            writer.WriteString("type", "boolean");
+        }
+        else if (value is int int32)
+        {
+            writer.WriteNumber("value", int32);
+            writer.WriteString("type", "integer");
+        }
+        else if (value is long int64)
+        {
+            writer.WriteNumber("value", int64);
+            writer.WriteString("type", "integer");
+        }
+        else if (value is double float64)
```

Comment:
I tried running this in C# interactive:
```
> var value = 13.3F;
> Console.WriteLine("{0}", value is double float64 ? float64 : null);
> (1,53): error CS8121: An expression of type 'float' cannot be handled by a pattern of type 'double'.
```

I know the [docs](https://develop.sentry.dev/sdk/telemetry/logs/#log-envelope-item-payload) only refer to double, but I think we should also match floats here (even if we send them over the wire with a `type` of `double`). It would be better than sending them with a type of `string`.

Reply:
Great catch!
I'll add more tests for more the numerics.